### PR TITLE
add filtering to OR for PSM checking

### PR DESCRIPTION
**Pull request by gadfort**
relates to https://github.com/siliconcompiler/zerosoc/pull/29

### CODE REVIEW

File: siliconcompiler/tools/openroad/scripts/sc_procs.tcl [hunk 1]
```diff
@@ -228,9 +244,32 @@ proc sc_supply_nets {} {
 
 proc sc_psm_check_nets {} {
   global openroad_psm_enable
+  global openroad_psm_skip_nets
 
   if { $openroad_psm_enable == "true" } {
-    return [sc_supply_nets]
+    set skip_nets []
+    foreach net_pattern $openroad_psm_skip_nets {
+      foreach skip_net [sc_find_net_regex $net_pattern] {
```
Comment (troibe) [inline siliconcompiler/tools/openroad/scripts/sc_procs.tcl:252]: Why do you first gather for the `skip_nets` list and then filter the `sc_supply_nets` using it?
Is it not possible to filter directly on the `sc_supply_nets`?

Comment (gadfort) [reply siliconcompiler/tools/openroad/scripts/sc_procs.tcl:252]: Add good point, I'll switch it around

File: siliconcompiler/tools/openroad/scripts/sc_procs.tcl [hunk 2]
```diff
@@ -228,9 +244,25 @@ proc sc_supply_nets {} {
...
+      foreach skip_pattern $openroad_psm_skip_nets {
+        if { [string match $skip_pattern $net] } {
+          set skipped true
+          break
+        }
+      }
+      if { !$skipped } {
+        lappend psm_nets $net
+      }
+    }
```
Comment (troibe) [inline siliconcompiler/tools/openroad/scripts/sc_procs.tcl:263]: Would something like this work? At least for full regexes.
For the `string match` I didn't find any pretty `lsearch` alternative...
```suggestion
    set joined_pattern [join $skip_nets "|"]
    foreach net [sc_supply_nets] {
      if { [regexp $joined_pattern $net] } {
        lappend psm_nets $net
      }
    }
```

**Review by troibe — APPROVED** [review]
Looks good to me overall.
Have a quick look at my suggestions before merging.

File: siliconcompiler/tools/openroad/scripts/sc_procs.tcl [hunk 3]
```diff
@@ -205,6 +205,22 @@ proc sc_bterm_has_placed_io { net } {
   return false
 }
 
+###########################
```
[the remark this answers is not in the document]
Comment (gadfort) [reply siliconcompiler/tools/openroad/scripts/sc_procs.tcl:208]: might be nice to have, so I'll leave it for now

File: siliconcompiler/tools/openroad/scripts/sc_procs.tcl (hunk 2 above)
Comment (gadfort) [reply siliconcompiler/tools/openroad/scripts/sc_procs.tcl:263]: regexp and string match are somewhat different and globbing is more common in EDA

**Pull request merged**